frontend/src/components/MilestoneCompletedness.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Trans } from 'react-i18next';
+import Tooltip from '@mui/material/Tooltip';
+import { Task } from '../redux/roadmaps/types';
+import { BarSection } from './PercentageBar';
+import { TaskStatus } from '../../../shared/types/customTypes';
+import { taskStatusToText } from '../utils/TaskUtils';
+import { percent } from '../utils/string';
+import colors from '../colors.module.scss';
+import css from './MilestoneCompletedness.module.scss';
+
+const classes = classNames.bind(css);
+
+const colorByStatus = (status: number) => {
+  if (status === TaskStatus.COMPLETED) return colors.emerald;
+  if (status === TaskStatus.IN_PROGRESS) return colors.orange;
+  return colors.black10;
+};
+
+export const MilestoneCompletedness: FC<{
+  tasks: Task[];
+}> = ({ tasks }) => {
+  const total = tasks.length;
+  const nums = tasks.reduce(
+    (acc, { status }) => ({ ...acc, [status]: acc[status] + 1 }),
+    {
+      [TaskStatus.COMPLETED]: 0,
+      [TaskStatus.IN_PROGRESS]: 0,
+      [TaskStatus.NOT_STARTED]: 0,
+    },
+  );
+  const orderedNums = Object.entries(nums).sort(
```

Comment:
```suggestion
  const orderedNums = Object.entries(nums).map(([k, v]) => [Number(k), v]).sort(
```

Could the statuses be converted back to numbers just once here